Add release notes for edge-19.6.3 (#2985)
linkerd/linkerd2: CHANGES.md

## edge-19.6.3

* CLI
  * Added a check for Pod Security Policy (PSP) permissions to `linkerd check`

>   * Changed `linkerd check` to only warn (not error) if PSPs are found, but

Reviewer: Nit/TIOLI: How about
```suggestion
  * Changed `linkerd check` to warn (rather than error) if PSPs are found, but
```

>     the caller does not have the `NET_ADMIN` capability

Reviewer 2: This isn't very clear to me... Can we summarize these two points as:

* Added `PodSecurityPolicy` warnings to `linkerd check`.

Is that an accurate summary of the change?

Reviewer 3: The task was to update `linkerd check` to ensure the caller has the RBAC permissions to create the control plane's PSP.

Reviewer 3: The 2nd bullet point is ok, but it isn't _the caller does not have_, it's _the existing PSPs do not have_. Thanks for calling it out. I missed that.

Reviewer 2: Perhaps:

* Updated `linkerd check` to validate the caller can create `PodSecurityPolicy` resources.

Author: I'll consolidate these two bullet points to the above. Thanks for the clarification

* Controller

>   * Improved timestamp formatting in logs

Reviewer 2: AIUI this only impacted development

  * Default the mutating and validating webhook configurations `sideEffects` 
    property to `None` to indicate that the webhooks have no side effects on

>     external resources

Reviewer 3: Sorry, my bad. I think we should replace _external_ with _other_ here, as the k8s doc seems to imply _all_ resources. Also, this is a contribution from Tarun.

* Proxy

>   * Added the `NET_RAW` capability to the proxy-init container to prevent it

Reviewer: Does this go in the proxy heading, or should there be a separate proxy-init heading? What have we done previously?

Author: I don't think we need to worry about introducing a separate heading for this. It could also go under "Controller", but I will probably keep it here.

>     from failing when the primary container has certain security contexts

Reviewer 3: I think a bit more specifics here will let users know whether this is relevant to them or not. So, _when the primary container has a `drop: ALL` capability security context_

>     from failing when the primary container has certain security contexts

Reviewer 2: replace "to prevent it from failing when ..." with "to satisfy `PodSecurityPolicy` requirements"

Reviewer 2: per @ihcsim, perhaps make this "to be compatible with `PodSecurityPolicy`s that use `drop: all`" ?

>   * Fixed the handling of HTTP2 requests that don't have an `:authority`

Reviewer: How about
```suggestion
  * Fixed the proxy rejecting HTTP2 requests that don't have an `:authority`
```

>   * Introduced eager eviction to the router cache so that idle services are

Reviewer: This seems like it could be made a little less wordy somehow? What about something like 
> Changed the proxy to eagerly evict idle services from router caches, even when they are not full

>     removed even when the cache is not full 

Reviewer 2: Let's make this a little more user-facing...

* Improved idle service eviction to reduce resource consumption for clients that send requests to many services.

Or something like that?

* Web UI

>   * Removed the "debug" page from the Linkerd dashboard while the functionality

Reviewer 4: Nit but "Debug" is capitalized in the dashboard so I'd capitalize here

    of that page is being redesigned
  * Reduced the dashboard package size and number of files loaded by importing

>     fewer icons

Reviewer 4: I think you can omit the bullet starting with "reduced the dashboard package size" as it's a small reduction in package size and not going to affect users
